cmd/service_linux.go

```diff
@@ -0,0 +1,335 @@
+		ServiceUser: serviceUser,
+	}
+
+	tmpl, err := template.New("service").Parse(serviceTemplate)
+	if err != nil {
+		fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	folderList := strings.Split(mountPath, "/")
+	serviceName := folderList[len(folderList)-1] + ".service"
+	newFile, err := os.Create("/etc/systemd/system/" + serviceName)
+	if err != nil {
+		return "", fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	err = tmpl.Execute(newFile, config)
+	if err != nil {
+		return "", fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	return serviceName, nil
+}
+
+func setUser(serviceUser string, mountPath string, configPath string) error {
+
+	configFileInfo, err := os.Stat(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to stat file: %v", err)
+	}
+
+	// Get file's group ID
+	stat, ok := configFileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return fmt.Errorf("failed to get file system stats")
+	}
+	configGroupID := stat.Gid
+
+	// Get configFileGroup name
+	configFileGroup, err := user.LookupGroupId(fmt.Sprint(configGroupID))
+	if err != nil {
+		return fmt.Errorf("failed to lookup group: %v", err)
+	}
+
+	mountPathInfo, err := os.Stat(mountPath)
+	if err != nil {
+		return fmt.Errorf("failed to stat file: %v", err)
+	}
+
+	// Get file's group ID
+	stat, ok = mountPathInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return fmt.Errorf("failed to get file system stats")
+	}
+	mountGroupID := stat.Gid
+
+	// Get configFileGroup name
+	mountPathGroup, err := user.LookupGroupId(fmt.Sprint(mountGroupID))
+	if err != nil {
+		return fmt.Errorf("failed to lookup group: %v", err)
+	}
+
+	_, err = user.Lookup(serviceUser)
+	if err != nil {
+		if strings.Contains(err.Error(), "unknown user") {
+			//create the user
+			userAddCmd := exec.Command("sudo", "useradd", "-m", serviceUser)
+			err = userAddCmd.Run()
+			if err != nil {
+				return fmt.Errorf("failed to create user due to following error: [%s]", err.Error())
+			}
+
+			//add group to serviceUser group
+			usermodCmd := exec.Command("sudo", "usermod", "-aG", configFileGroup.Name, serviceUser)
+			err = usermodCmd.Run()
+			if err != nil {
+				return fmt.Errorf("failed to create user due to following error: [%s]", err.Error())
+			}
+			usermodCmd = exec.Command("sudo", "usermod", "-aG", mountPathGroup.Name, serviceUser)
+			err = usermodCmd.Run()
+			if err != nil {
+				return fmt.Errorf("failed to create user due to following error: [%s]", err.Error())
+			}
+
+			//set set folder permission on the mount path
+			chmodCmd := exec.Command("sudo", "chmod", "770", mountPath)
+			err = chmodCmd.Run()
+			if err != nil {
+				return fmt.Errorf("failed set permisions on mount path due to following error: [%s]", err.Error())
+			}
+
+		} else {
+			fmt.Printf("An error occurred: %v\n", err)
+		}
+	} else {
+		//add group to serviceUser group
+		usermodCmd := exec.Command("sudo", "usermod", "-aG", configFileGroup.Name, serviceUser)
```

Comment:
If the user exists, we probably should not modify the group permissions and assume the user has them set in a way to run cloudfuse. And if there is an error we should tell them how to fix that error.